### ISSUE TITLE
[CT-1218] [Bug] The query-comment statement not included in incremental Merge query

### ISSUE THREAD

**Issue by pempey** [dbt-labs/dbt-snowflake#270]
### Is this a new bug in dbt-snowflake?

- [X] I believe this is a new bug in dbt-snowflake
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

When reviewing the query text in snowflake for incremental model executions the `query-comment` is not present in the `MERGE` command but it is present in the `CREATE` temp table command.

### Expected Behavior

I would expect all queries executed by dbt to include the defined `query-comment`

### Steps To Reproduce

1. dbt-snowflake version 1.1
2. define a query-comment in the dbt-project.yml
3. create an incremental model
4. run the incremental model first to build the table and a second time to invoke a merge action.
5. review dbt logs or snowflake query logs

### Environment

```markdown
- OS:macOS 11.6.6
- Python: 3.8.6
- dbt-core: 1.1.0
- dbt-snowflake: 1.1.0
```

**Comment by jtcohen6** [comment]
Hey @pempey - I believe this was fixed (https://github.com/dbt-labs/dbt-snowflake/pull/153)!

That fix was included in v1.2 — could you try upgrading, and see whether you see the query comment show up on the `merge` statement as well?

**Comment by pempey** [comment]
@jtcohen6 I can confirm that upgrading to v1.2. solved the issue.  Not sure why I did not see the other issues but thank you for helping me resolve this.
